src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -273,20 +278,18 @@ static SFLoginOutput openSession(
     AssertUtil.assertTrue(
         loginInput.getLoginTimeout() >= 0, "negative login timeout for opening session");
 
-    if (getAuthenticator(loginInput).equals(AuthenticatorType.OAUTH_AUTHORIZATION_CODE)) {
-      AssertUtil.assertTrue(
-          loginInput.getOauthLoginInput().getClientId() != null,
-          "passing clientId is required for OAUTH_AUTHORIZATION_CODE_FLOW authentication");
-      AssertUtil.assertTrue(
-          loginInput.getOauthLoginInput().getClientSecret() != null,
-          "passing clientSecret is required for OAUTH_AUTHORIZATION_CODE_FLOW authentication");
-      OauthAccessTokenProvider accessTokenProvider =
-          new AuthorizationCodeFlowAccessTokenProvider(
+    if (AccessTokenProviderFactory.isEligible(getAuthenticator(loginInput))) {
+      AccessTokenProviderFactory accessTokenProviderFactory =
+          new AccessTokenProviderFactory(
               new SessionUtilExternalBrowser.DefaultAuthExternalBrowserHandlers(),
               (int) loginInput.getBrowserResponseTimeout().getSeconds());
+      AccessTokenProvider accessTokenProvider =
+          accessTokenProviderFactory.createAccessTokenProvider(
+              getAuthenticator(loginInput), loginInput);
       String oauthAccessToken = accessTokenProvider.getAccessToken(loginInput);
       loginInput.setAuthenticator(AuthenticatorType.OAUTH.name());
       loginInput.setToken(oauthAccessToken);
+      loginInput.setUserName("0oalpyiuy8rmozhjZ5d7");
```

Comment:
looks like the secret and git guardian should complain

Reply:
It doesn't though 🤷  Removed it.